### PR TITLE
fix(client): Handle old accounts that contain `accountData`

### LINKED ISSUE #3466: Reported error: accountData cannot be set on an Account
See report here: https://sentry.prod.mozaws.net/operations/fxa-content-product/group/223558/events/5923791/

```
/scripts/main.js  ? in s<.set< at line 32:19954
/scripts/main.js  ? in D at line 13:12921
/scripts/main.js  ? in s.partial/c at line 13:13444
/scripts/main.js  ? in b.Model at line 16:31774
/scripts/main.js  ? in J/d< at line 17:15787
/scripts/main.js  ? in l<.initAccount at line 32:22921
/scripts/main.js  ? in l<.getSignedInAccount at line 32:23293
/scripts/main.js  ? in l<.upgradeFromSession/< at line 32:25011
/scripts/main.js  ? in F at line 17:23170
/scripts/main.js  ? in E at line 17:23124
/scripts/main.js  ? in l at line 17:21172
```

image

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Upgrade storage formats before trying to do any more data access.

fixes #3466

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@vladikoff - mind an r?

**Comment by philbooth** [comment]
Tested locally and works for me, looks good too. :+1:

File: app/scripts/lib/app-start.js [hunk 1]
```diff
@@ -519,7 +519,14 @@ define(function (require, exports, module) {
     },
 
     upgradeStorageFormats: function () {
-      return this._user.upgradeFromSession(Session, this._fxaClient);
+      var user = this._user;
+
+      // upgradeFromUnfilteredAccountData comes first
+      // otherwise upgradeFromSession fails because it tries
+      // to read and create Accounts for unfiltered data.
```
Comment (vladikoff) [inline app/scripts/lib/app-start.js]: `for` -> `from` ?

File: app/scripts/lib/app-start.js [hunk 2]
```diff
@@ -519,7 +519,14 @@ define(function (require, exports, module) {
...
+      // upgradeFromSession writes the new format, so this is safe.
```
Comment (vladikoff) [inline app/scripts/lib/app-start.js]: nit: `so this is safe.` sounds ambiguous

File: app/scripts/models/user.js [hunk 3]
```diff
@@ -290,6 +290,27 @@ define(function (require, exports, module) {
         });
     },
 
+    // Before a13f05f2 (18 Dec 2014), all kinds of extra
+    // data was written to the Account. This extra data hung
+    // arround even if the user signed in again. After d4321990
+    // (12 Jan 2016), only allowed fields are allowed to be
+    // set on an account, unexpected fields cause an error.
```
Comment (vladikoff) [inline app/scripts/models/user.js]: nit: `set on an account,` -> `set on an Account,` ?

**Comment by vladikoff** [comment]
r+ with nits :)

**Pull request merged**